src/main/java/org/openstreetmap/atlas/utilities/archive/Extractor.java

```diff
@@ -187,6 +179,12 @@ public Extractor extract(final File inputFile) throws ArchiveException, IOExcept
             {
                 final File outputFile = new File(this.outputDirectory, current.getName());
 
+                if (current.getName().contains(".."))
```

Comment:
This solves the vulnerability by checking for parent folder injection... Not sure if it is the best way.